dds/DCPS/transport/shmem/ShmemDataLink.h

```diff
@@ -28,21 +29,18 @@ namespace DCPS {
 class ReceivedDataSample;
 
 struct ShmemData {
-  int status_;
+  enum Status {
+    Free = 0,
+    InUse = 1,
+    RecvDone = 2,
+    EndOfAlloc = -1
+  };
+
+  Status status_;
```

Comment:
```suggestion
  int status_;
```
This was chosen to make the in-memory representation independent of compiler/implementation decisions.
We can't guarantee that two processes running code built with different compilers can communicate over shmem, but we'll try to support it when possible.

Reply:
If that's the intention, wouldn't it be better if this was using something like `ACE_UINT8`?